SingularityUI/app/components/common/modal/FormModal.jsx

```diff
@@ -119,20 +114,24 @@ export default class FormModal extends React.Component {
     const parsed = {};
     _.mapObject(state, (val, key) => {
       const element = _.find(this.props.formElements, (formElement) => formElement.name === key);
-      switch (element.type) {
-        case FormModal.INPUT_TYPES.BOOLEAN:
-          parsed[key] = Boolean(val);
-          break;
-        case FormModal.INPUT_TYPES.NUMBER:
-          parsed[key] = Number.parseFloat(val);
-          break;
-        case FormModal.INPUT_TYPES.DURATION:
-          if (val) {
-            parsed[key] = juration.parse(val) * 1000;
-          }
-          break;
-        default:
-          parsed[key] = val;
+      if (element === undefined) {
+        delete this.state.formState[key];
```

Comment:
we don't want to directly manipulate the state here. Instead we need to eventually pass the new data to `setState`. We use the return value from this method to set the state correctly on line 153. So, I think if the element is undefined we can just skip that element and move on. It will then not end up in `parsed` and will be later removed from the state when we do `setState`